src/github.com/getlantern/flashlight/statserver/peer.go

```diff
@@ -50,8 +50,9 @@ func newPeer(ip string, pub publish) *Peer {
 
 func (peer *Peer) run() {
 	err := peer.geolocate()
+
 	if err != nil {
-		log.Errorf("Unable to geolocate peer after %d attempts, stopping reporting: %v", maxGeolocateTries, err)
+		log.Errorf("Unable to geolocate peer after %d attempts, stopping reporting. Last error was: %q", maxGeolocateTries, err)
```

Comment:
The error message did not make clear the logged error was the last one of a series of attempts.